x-pack/plugins/security_solution/public/management/pages/policy/view/artifacts/list/policy_artifacts_list.tsx

```diff
@@ -0,0 +1,186 @@
+      | Partial<HostIsolationExceptionsPageLocation>
+  ) => string;
+  getPolicyArtifactsPath: (policyId: string) => string;
+  labels: typeof POLICY_ARTIFACT_LIST_LABELS;
+  onDeleteActionCallback: (item: ExceptionListItemSchema) => void;
+  externalPrivileges?: boolean;
+}
+
+export const PolicyArtifactsList = React.memo<PolicyArtifactsListProps>(
+  ({
+    policy,
+    apiClient,
+    searcheableFields,
+    getArtifactPath,
+    getPolicyArtifactsPath,
+    labels,
+    onDeleteActionCallback,
+    externalPrivileges = true,
+  }) => {
+    const { getAppUrl } = useAppUrl();
+    const { canCreateArtifactsByPolicy } = useUserPrivileges().endpointPrivileges;
+    const policiesRequest = useGetEndpointSpecificPolicies({ perPage: 1000 });
+    const navigateCallback = usePolicyDetailsArtifactsNavigateCallback(apiClient.listId);
+    const { urlParams } = useUrlParams();
+    const [expandedItemsMap, setExpandedItemsMap] = useState<Map<string, boolean>>(new Map());
+
+    const { state } = useGetLinkTo(policy.id, policy.name, getPolicyArtifactsPath, getArtifactPath);
+
+    const {
+      data: artifacts,
+      isLoading: isLoadingArtifacts,
+      isRefetching: isRefetchingArtifacts,
+    } = useListArtifact(apiClient, searcheableFields, {
+      page: Number(urlParams.page_index) + 1 || undefined,
+      perPage: Number(urlParams.page_size) || undefined,
+      filter: urlParams.filter as string,
+      policies: [policy.id, 'all'],
+    });
+
+    const pagination: Pagination = {
+      pageSize: Number(urlParams.page_size) || MANAGEMENT_DEFAULT_PAGE_SIZE,
+      pageIndex: Number(urlParams.page_index) || 0,
+      pageSizeOptions: [...MANAGEMENT_PAGE_SIZE_OPTIONS],
+      totalItemCount: artifacts?.total || 0,
+    };
+
+    const handleOnSearch = useCallback(
+      (filter) => {
+        navigateCallback({ filter });
+      },
+      [navigateCallback]
+    );
+
+    const handleOnExpandCollapse: ArtifactCardGridProps['onExpandCollapse'] = ({
+      expanded,
+      collapsed,
+    }) => {
+      const newExpandedMap = new Map(expandedItemsMap);
+      for (const item of expanded) {
+        newExpandedMap.set(item.id, true);
+      }
+      for (const item of collapsed) {
+        newExpandedMap.set(item.id, false);
+      }
+      setExpandedItemsMap(newExpandedMap);
+    };
+    const handleOnPageChange = useCallback<ArtifactCardGridProps['onPageChange']>(
+      ({ pageIndex, pageSize }) => {
+        if (artifacts?.total) navigateCallback({ page_index: pageIndex, page_size: pageSize });
+      },
+      [artifacts?.total, navigateCallback]
+    );
+
+    const totalItemsCountLabel = useMemo<string>(() => {
+      return labels.listTotalItemCountMessage(artifacts?.data.length || 0);
+    }, [artifacts?.data.length, labels]);
+
+    const artifactCardPolicies = useEndpointPoliciesToArtifactPolicies(policiesRequest.data?.items);
+    const provideCardProps: ArtifactCardGridProps['cardComponentProps'] = (artifact) => {
```

Comment:
I have a hook in my `ArtifactListPage` that provides this. Wondering if we should promote it and just reuse here?

https://github.com/elastic/kibana/blob/f4a13a883b71c53aba8663ee52c33091375d5d83/x-pack/plugins/security_solution/public/management/components/artifact_list_page/hooks/use_artifact_card_props_provider.ts

Reply:
Is this one `useArtifactCardPropsProvider`? I think we should do a bit of refactor then to add the privilege checks conditions.

Reply:
yeah, that one. Let's hold off on using them, since some refactoring is needed.